### PR TITLE
Rework content on view job page

### PR DESCRIPTION
**Pull request by AntoineAugusti**
Rework https://github.com/cds-snc/notification-admin/pull/1066, some content was a bit off. Add more tests.

### CODE REVIEW

File: tests/app/main/views/test_jobs.py [hunk 1]
```diff
@@ -409,6 +410,60 @@ def test_should_show_scheduled_job(
...
+):
+    page = client_request.get(
+        "main.view_job",
+        service_id=SERVICE_ONE_ID,
+        job_id=fake_uuid,
+    )
+
+    assert normalize_spaces(page.select("main p")[0].text) == (
+        "'Two week reminder' scheduled from the API using the key 'API key name'"
+    )
```
Comment (jimleroyer) [inline tests/app/main/views/test_jobs.py:450]: Is this assertion above testing the same functionality as in the previous test?

File: tests/conftest.py [hunk 2]
```diff
@@ -2045,6 +2045,14 @@ def _get_job(service_id, job_id):
     return mocker.patch("app.job_api_client.get_job", side_effect=_get_job)
 
 
+@pytest.fixture(scope="function")
+def mock_get_job_with_api_key(mocker, api_user_active):
+    def _get_job(service_id, job_id):
+        return {"data": job_json(service_id, api_user_active, job_id=job_id, api_key={"name": "API key name"})}
```
Comment (jimleroyer) [inline tests/conftest.py:2051]: Oh here is the `API key name`. Is this worth it to make it a constant if we test it in assertions?

**Comment by AntoineAugusti** [comment]
@jimleroyer Introduced a constant for the job API key name

File: tests/app/main/views/test_jobs.py (hunk 1 above)
Comment (AntoineAugusti) [reply tests/app/main/views/test_jobs.py:451]: Almost! This test is for a scheduled job from the API, the previous is for a job that's not scheduled from the API
> Two week reminder' sent from the API using the key 'API key name' on 2016-01-01T00:00:00.061258+0000

VS 

> 'Two week reminder' **scheduled** from the API using the key 'API key name'

File: app/templates/partials/jobs/status.html [hunk 3]
```diff
@@ -2,25 +2,28 @@
...
         {% endif %}
       {% else %}
         {% if not job.api_key %}
           {{ _('Uploaded by') }} {{ job.created_by.name }} {{ _('on') }} <span class="local-datetime-short">{{ job.created_at}}</span>
-          }
         {% else %}
-          {{ _("'<a href='{}'>{}</a>' scheduled from the API using the key <a href='{}'>{}</a>").format(url_for('.view_template', service_id=current_service.id, template_id=template.id), template.name, url_for('.api_keys', service_id=current_service.id), job.api_key.name) }}
+          {{ _("'<a href='{}'>{}</a>' scheduled from the API using the key '<a href='{}'>{}</a>'").format(url_for('.view_template_version', service_id=current_service.id, template_id=template.id, version=template_version), template.name, url_for('.api_keys', service_id=current_service.id), job.api_key.name) }}
```
[the remark this answers is not in the document]
Comment (AntoineAugusti) [reply app/templates/partials/jobs/status.html:19]: There is!

File: tests/app/main/views/test_jobs.py [hunk 4]
```diff
@@ -409,6 +410,60 @@ def test_should_show_scheduled_job(
...
+    fake_uuid,
+):
+    page = client_request.get(
+        "main.view_job",
+        service_id=SERVICE_ONE_ID,
+        job_id=fake_uuid,
+    )
+
+    assert normalize_spaces(page.select("main p")[0].text) == (
+        "'Two week reminder' sent from the API using the key 'API key name' on 2016-01-01T00:00:00.061258+0000"
```
[the remark this answers is not in the document]
Comment (AntoineAugusti) [reply tests/app/main/views/test_jobs.py]: Done in the next commit

File: tests/conftest.py (hunk 2 above)
Comment (AntoineAugusti) [reply tests/conftest.py]: Done in the next commit

**Review by jimleroyer — APPROVED** [review]

**Pull request merged**